mkall/MKCollector.h

```diff
@@ -33,8 +33,21 @@
 @interface MKCollectorResubmitTask : NSObject
 
 /// initWithSerializedMeasurement initializes the task using the
-/// serialized measurement that should be uploaded to the OONI collector.
--(instancetype)initWithSerializedMeasurement:(NSString *)content;
+/// serialized measurement that should be uploaded to the OONI collector,
+/// the software name and the software version of the app.
+-(instancetype)initWithSerializedMeasurement:(NSString *)content
+                                softwareName:(NSString *)softwareName
+                             softwareVersion:(NSString *)softwareVersion;
```

Comment:
@lorenzoPrimi the changes you see here are based on previous comments from @xan:

1. it is weird to call `FooSettings` a class that has a `perform` method, hence call it `FooTask`

2. it is more clean OO to have the constructor take as mandatory arguments the arguments that actually are required to properly construct a class

Reply:
I see, so I should make these changes everywhere in the app, right?

Reply:
Yes